aopy/utils/base.py

```diff
@@ -765,4 +765,39 @@ def convert_port_number(port_number, datatype='ap'):
     letter = chr(ord('@')+port_number)
     probe_dir = f'Neuropix-PXI-100.Probe{letter}-{datatype.upper()}'
     
-    return probe_dir
\ No newline at end of file
+    return probe_dir
+
+def multiply_mat_batch(data, mat, save_path, sample_rate, scale = 1, batch_length = 2, dtype='int16'):
+    '''
+    Multiply a matrix to data in each batch to save memory. The result is saved in save_path.
+    This function can be used to multiply an inverse matrix by spike band time series.
+    
+    Args:
+        data (nt, nch): neural data. This should be a memory mapping array.
+        mat (anysize, nch): matrix to multiply by data
+        save_path (str): file path to save destriped lfp data
+        sample_rate (float): sampling rate in Hz
+        scale (float, optional): Scaling factor to multiply by data. 1/200 is necessary for whitened data in kilosort4. default is 1.
+        batch_length (int, optional): batch size in second. default is 2 seconds.
```

Comment:
Similar to the other pull request, I think there should at least be an option to define the batch size based on available memory.